cachito/web/content_manifest.py

```diff
@@ -43,8 +46,10 @@ def process_gomod(self, package, dependency):
         :param Dependency dependency: the gomod package dependency to process
         """
         if dependency.type == "gomod":
-            icm_source = {"purl": dependency.to_purl()}
-            self._gomod_data[package.name].append(icm_source)
+            parent_purl = self._gomod_data[package.name]["purl"]
+            dep_purl = dependency.to_purl().replace(PARENT_PURL_PLACEHOLDER, parent_purl)
+            icm_source = {"purl": dep_purl}
+            self._gomod_data[package.name]["dependencies"].append(icm_source)
```

Comment:
nitpick: It would be nice if the structure change (using `"dependencies"`) was in a separate commit here.

Reply:
Broke up into 2 commits